### PR TITLE
Timestamps, JUnit Tests, Comments // Implemented

### PR DESCRIPTION
**Pull request by philip1304**
Notes:
- Added TimestampIO.java & TimestampIOTest.java
- Added further JUnit tests in HelpCommandTest.java
- Added in-line comments to StringMatcher.java for method clarity

### CODE REVIEW

File: src/main/java/seedu/pill/util/StringMatcher.java [hunk 1]
```diff
@@ -19,23 +19,32 @@ public class StringMatcher {
...
-                            dp[i - 1][j] + 1,
-                            dp[i][j - 1] + 1);
+                    // Calculate minimum cost for current position using three possible operations:
+                    dp[i][j] = min(
+                        // Substitution (or no change if characters are same)
+                        dp[i - 1][j - 1] + (s1.charAt(i - 1) == s2.charAt(j - 1) ? 0 : 1),
+                        // Deletion from s1
+                        dp[i - 1][j] + 1,
+                        // Insertion into s1
+                        dp[i][j - 1] + 1);
```
Comment (yijiano) [inline src/main/java/seedu/pill/util/StringMatcher.java:42]: Good implementation of DP.

File: src/test/java/seedu/pill/util/TimestampIOTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class TimestampIOTest {
```
Comment (yijiano) [inline src/test/java/seedu/pill/util/TimestampIOTest.java:15]: Perhaps more failure cases could have been added.

**Review by yijiano — APPROVED** [review]
LGTM

File: src/test/java/seedu/pill/util/TimestampIOTest.java (hunk 2 above)
Comment (philip1304) [reply src/test/java/seedu/pill/util/TimestampIOTest.java:15]: Will do soonTM.

**Pull request merged**